timely/src/progress/broadcast.rs

```diff
@@ -138,32 +131,31 @@ impl<T:Timestamp+Send> Progcaster<T> {
 
             // See comments above about the relatively high cost of this logging, and our
             // options for improving it if performance limits users who want other logging.
-            self.logging.as_ref().map(|l| {
+            self.progress_logging.as_ref().map(|l| {
 
                 let mut messages = Vec::with_capacity(changes.len());
                 let mut internal = Vec::with_capacity(changes.len());
 
-                // TODO: Reconsider `String` type or perhaps re-use allocation.
                 for ((location, time), diff) in recv_changes.iter() {
 
                     match location.port {
                         Port::Target(port) => {
-                            messages.push((location.node, port, format!("{:?}", time), *diff))
+                            messages.push((location.node, port, time.clone(), *diff))
                         },
                         Port::Source(port) => {
-                            internal.push((location.node, port, format!("{:?}", time), *diff))
+                            internal.push((location.node, port, time.clone(), *diff))
                         }
                     }
                 }
 
-                l.log(crate::logging::ProgressEvent {
+                l.log(crate::logging::TimelyProgressEvent {
                     is_send: false,
                     source: source,
                     seq_no: counter,
                     channel,
                     addr: addr.clone(),
-                    messages,
-                    internal,
+                    messages: Box::new(messages),
+                    internal: Box::new(internal),
```

Comment:
Yeah, this is what I thought you would need to do. Is there a reason this is needed here but not previously? If so I think we should doc it, and if not we should unify up the two cases to look the same (maybe just aesthetics, but also to avoid randoms (myself) asking this question in the future).

Reply:
The asymmetry was unintentional (and I guess a potential perf defect). Fixed in the last commit.